runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt

```diff
@@ -51,7 +51,7 @@ public class PolymorphicModuleBuilder<Base : Any> internal constructor(
      * @see addSubclass
      */
     @ImplicitReflectionSerializer
-    public inline fun <reified T : Base> addSubclass(): Unit = addSubclass(T::class, T::class.serializer())
+    public inline fun <reified T : Base> addSubclass(): Unit = addSubclass(T::class, serializer())
```

Comment:
maybe better rename this function to `subclass` alone:

```kotlin
polymorphic(Any::class, PolyBase::class) {
  subclass<PolyDerived>() // Looks more DSL-ish
}
```

Reply:
Great catch!